nt-be/migrations/20251223000001_create_balance_changes.sql

```diff
@@ -0,0 +1,52 @@
+-- Create balance_changes table for storing account balance history
+CREATE TABLE balance_changes (
+    id BIGSERIAL PRIMARY KEY,
+    account_id VARCHAR(64) NOT NULL,
+    
+    -- Block metadata
+    block_height BIGINT NOT NULL,
+    block_timestamp BIGINT NOT NULL,
+    
+    -- Transaction info
+    transaction_block BIGINT,
+    transaction_hashes TEXT[] NOT NULL DEFAULT '{}',
+    signer_id VARCHAR(64),
+    receiver_id VARCHAR(64),
+    
+    -- Token and transfer data
+    token_id VARCHAR(64),
+    receipt_id TEXT[] NOT NULL DEFAULT '{}',
+    counterparty VARCHAR(64) NOT NULL,
+    amount NUMERIC(78, 0) NOT NULL,  -- Up to 78 digits for yoctoNEAR amounts
+    balance_before NUMERIC(78, 0) NOT NULL,  -- Arbitrary precision for large token amounts
+    balance_after NUMERIC(78, 0) NOT NULL,  -- Arbitrary precision for large token amounts
+    
+    -- Raw data for debugging
+    actions JSONB,
+    raw_data JSONB,
+    
+    -- Timestamps
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    
+    -- Constraints
+    CONSTRAINT unique_account_block_token UNIQUE(account_id, block_height, token_id),
```

Comment:
Any chance that user might receive N transfers across the block? Or do you plan to aggregate?